### DISCUSSION TITLE
Best tool for ID3 debugging in HLS

### DISCUSSION THREAD

**Discussion by frant1** [savonet/liquidsoap#4176] [User Support]
Hello,

I'm currently doing an HLS stream with Liquidsoap and I'm looking for adding ID3 to my stream but I don't really understand how to debug that stuff. Usually VLC is the best for everything regarding medias but this time it won't help me as it don't shows the metadatas..

I saw with HLS.JS sandbox I could have some results but as it's web based I still have problem with caches.. Would there be a windows app that can monitor it ?

Regards,
Antoine

**Comment by toots** [comment]
Hi! I've been using hls.js. `ffmpeg` does not have great support for that. 

What format are you using? For `mpegts` the metadata is inside a data stream, you need to be able to manually inspect it. For `adts`/`mp3`, it's a simple ID3 tag at the beginning of the segments. This is one might be easier to manually catch.

**Comment by Russsgithub** [comment]
@frant1 , there's also a fork of just audio flutter lib

https://github.com/tadevel/just_audio/tree/hls_metadata

That has hls id3 metadata support , which you can pass Json as say the title / artist which works. It has not been accepted by the Dev to be merged but it does work.
A bit of a large install and build process, but an option for use through the emulator or build for phone etc.
